src/utils/SignatureCollectionMap.ts

```diff
@@ -7,49 +7,58 @@ type SignerAndSignature = {
 };
 
 export class SignatureCollectionMap {
-    private map: Map<string, SignerAndSignature[]> = new Map<
-        string,
-        SignerAndSignature[]
-    >();
+    // Replace array with a nested Map: signerAddress -> signature
+    private map: Map<string, Map<string, SignatureLike>> = new Map();
 
     public tryInsert(key: string, value: SignerAndSignature): void {
         if (!this.map.has(key)) {
-            this.map.set(key, [value]);
+            this.map.set(key, new Map());
         }
-        let array = this.get(key);
-        for (let i = 0; i < array.length; i++) {
-            if (array[i].signerAddress === value.signerAddress) {
-                return;
-            }
+
+        // Get the inner map and add the signature if not already present
+        const innerMap = this.map.get(key)!;
+        if (!innerMap.has(value.signerAddress)) {
+            innerMap.set(value.signerAddress, value.signature);
         }
-        array.push(value);
     }
+
     public didEveryoneSign(key: string, participants: AddressLike[]): boolean {
-        let array = this.get(key);
-        for (let i = 0; i < participants.length; i++) {
-            let found = false;
-            for (let j = 0; j < array.length; j++) {
-                if (array[j].signerAddress === participants[i].toString()) {
-                    found = true;
-                    break;
-                }
-            }
-            if (!found) {
-                return false;
-            }
-        }
-        return true;
+        const innerMap = this.map.get(key);
+        if (!innerMap) return false;
+
+        // Check if every participant has a signature
+        return participants.every((participant) =>
+            innerMap.has(participant.toString())
+        );
     }
+
+    // Return in the same format as before for backward compatibility
```

Comment:
why Backwards compatibility ? Do we already have things in prod ?

Reply:
no need indeed
fixed https://github.com/peer3to/state-channels-plus/pull/48/commits/23b5ab3cc83505d6f0a84d5ff00ebbabe957fee5